packages/api/models/collections.js

```diff
@@ -85,6 +87,17 @@ const deprecatedFields = Object.freeze([
   'provider_path',
 ]);
 
+class CollectionManager extends Manager {
```

Comment:
I'm not sure I understand why we need this class? I think that `addTimeStampsToItem` needs to be defined on the collection model itself in order to have any effect?

Reply:
I'm still not following. In the providers model, for example, `addTimeStampsToItem` is defined directly on the model class. That makes sense because it's trying to provide an override to a base class method. However, in this case, you're defining `addTimeStampsToItem` on the object which becomes `this.dynamoDbClient` in the collections model, so I'm not seeing how the base class override ever happens?

Reply:
I believe that's due to this class overriding all that with the following: 
`
  async createItem(item) {
    validateCollection(item);

    const { name, version } = item;
    await this.collectionConfigStore.put(name, version, item);

    return this.dynamoDbClient.create(item);
  }

  createItems(items) {
    return Promise.all(
      items.map((item) => this.createItem(item))
    );
  }

  create(input) {
    if (Array.isArray(input)) {
      return this.createItems(input);
    }

    return this.createItem(input);
  }
  `
  
  ^ The above uses the custom CollectionManager (that has the `.create method` that uses the override)  that's defined in the class constructor as 

`  
  this.dynamoDbClient = new CollectionManager({
`

I was, perhaps in retrospect, to ready to not refactor this in light of upcoming rewrites.

Reply:
I need to verify this in a deployed stack, but reordering all of them appears to work in context of Core: 

476ccf143